src/mpid/ch3/channels/sock/src/ch3_progress.c

```diff
@@ -134,29 +134,6 @@ static int MPIDI_CH3i_Progress_wait(MPID_Progress_state * progress_state)
     }
 #endif
 
-#ifdef MPICH_IS_THREADED
-    MPIR_THREAD_CHECK_BEGIN;
-    {
-        if (MPIDI_CH3I_progress_blocked == TRUE) {
-            /*
-             * Another thread is already blocking in the progress engine.
-             *
-             * MT: Another thread is already blocking in poll.  Right now,
-             * calls to MPIDI_CH3_Progress_wait() are effectively
-             * serialized by the device.  The only way another thread may
-             * enter this function is if MPIDI_CH3I_Sock_wait() blocks.  If
-             * this changes, a flag other than MPIDI_CH3I_Progress_blocked
-             * may be required to determine if another thread is in
-             * the progress engine.
-             */
-            MPIDI_CH3I_Progress_delay(MPIDI_CH3I_progress_completion_count);
```

Comment:
Do progress hooks provide their own safety? I don't get how this can work.

Reply:
`MPIR_THREAD_CHECK_BEGIN/END` is simply checking whether we are single-threaded or multi-threaded. It is not to do with thread-safety. I'll need refresh my memory on what the change is doing... I'll add another comment.

Reply:
In `ch3_sock`, the progress_wait enters blocking `poll` or sleep on a condition_var. So we need make sure only enter `poll` or sleep when we expect some network events will come. With nonblocking collective, sometime all the network traffic is done, but the request isn't complete simply because the `sched` hook haven't run yet. So we need ensure the hook gets run and checked before we block (sleep in this case).